src/main/java/ca/uhn/fhir/jpa/starter/CustomAuthorizationInterceptor.java

```diff
@@ -13,35 +23,107 @@
 
 @Interceptor
 public class CustomAuthorizationInterceptor extends AuthorizationInterceptor {
-
-	private static final String HEADER_NAME = "x-api-key";
-	private static final String VALIDATE_API_KEY = "VALIDATE_API_KEY";
-	private static final String API_KEY = "API_KEY";
+	private static final Logger logger = LoggerFactory.getLogger(CustomAuthorizationInterceptor.class);
+	private static final String OAUTH_URL = System.getenv("OAUTH_URL");
+	private static final String APIKEY_ENABLED = System.getenv("APIKEY_ENABLED");
+	private static final String APIKEY_HEADER = "x-api-key";
+	private static final String APIKEY = System.getenv("APIKEY");
+	private static final String FHIR_VERSION = System.getenv("fhir_version");
+	private static final String TOKEN_PREFIX = "BEARER ";
+	private static PublicKey publicKey = null;
+	private static OAuth2Helper oAuth2Helper = new OAuth2Helper();
 
 	@Override
-	public List<IAuthRule> buildRuleList(RequestDetails theRequestDetails) {
+	public List<IAuthRule> buildRuleList(RequestDetails theRequest) {
 		try {
 
-			boolean validateApiKey = Boolean.parseBoolean(System.getenv(VALIDATE_API_KEY));
-			String token = theRequestDetails.getHeader(HEADER_NAME);
+			if (theRequest.getRequestPath().equals(RestOperationTypeEnum.METADATA.getCode())) {
+				return allowAll();
+			}
+
+			if (!oAuth2Helper.isOAuthEnabled() && !isApiKeyEnabled()) {
+				logger.info("Apikey & Oauth2 are disabled");
```

Comment:
Please change this to a warning and make the message "APIKEY and OAuth2 authentication are disabled".